internal/handlers/common/projection.go

```diff
@@ -121,8 +159,8 @@ func ProjectDocuments(docs []*types.Document, projection *types.Document) error
 func projectDocument(inclusion bool, doc *types.Document, projection *types.Document) error {
 	projectionMap := projection.Map()
 
-	for k1 := range doc.Map() {
-		projectionVal, ok := projectionMap[k1]
+	for k1, k1Val := range doc.Map() {
```

Comment:
Is there any specific reason why it's `k1, k1Val` and not just `key, val`? I think it might be easier to read.

Reply:
it is just 1 and 2 level of `elemMatch` it's easier to understand. 
Nested `elemMatch` will be in other PR.